cmd/plugins/balloons/policy/balloons-policy.go

```diff
@@ -399,16 +399,27 @@ func (p *balloons) balloonDefByName(defName string) *BalloonDef {
 }
 
 func (p *balloons) chooseBalloonDef(c cache.Container) (*BalloonDef, error) {
-	var blnDef *BalloonDef
 	// BalloonDef is defined by annotation?
 	if blnDefName, ok := c.GetEffectiveAnnotation(balloonKey); ok {
-		blnDef = p.balloonDefByName(blnDefName)
+		blnDef := p.balloonDefByName(blnDefName)
 		if blnDef == nil {
 			return nil, balloonsError("no balloon for annotation %q", blnDefName)
 		}
 		return blnDef, nil
 	}
 
+	// BalloonDef is defined by a match expression?
+	for _, blnDef := range append([]*BalloonDef{p.reservedBalloonDef, p.defaultBalloonDef}, p.bpoptions.BalloonDefs...) {
```

Comment:
I agree. It would be nice to get balloon (def.) precedence follow the order of definition of balloons in the configuration, from highest to lowest, with as few exceptions and other surprising side-effects as possible.

> Or if we happen to merge #254 first, then lets change this order while resolving the conflict.

Shortest straw loses ? :see_no_evil: